src/cluster_legacy.c

```diff
@@ -6890,9 +6913,26 @@ void clusterPromoteSelfToMaster(void) {
     replicationUnsetMaster();
 }
 
+int detectAndUpdateCachedNodeHealth(void) {
```

Comment:
Can we return the number of nodes whose health changed and consider the func name below, for clearer contract and readability? 

```suggestion
int updateAndCountChangedNodeHealth(void) {
```

Reply:
Why is this clearer? It just seems different.

Reply:
Yes, It would be a different purpose, which we do not require to invalidate the cached response. even one one change detection would suffice.

Reply:
> Why is this clearer? It just seems different.

I can't easily tell what `detectAndUpdateCachedNodeHealth` is trying to achieve without looking into the implementation. We could keep the current implementation but I would prefer a name that is more "boolean'ish", which would be hard to name IMO (`detectAndUpdateCachedNodeHealthAndCheckIfThereIsAChange`? :-)) - the fact that we can't use a bool type also makes the function contract less clear btw. The function name would be more self-explanatory if we go with `updateAndCountChangedNodeHealth`, which is why I suggested "counting".

Reply:
Maybe I don't see this as confusing, as the method name seems to be self descriptive. I'm ok with this change as is. @PingXie If it's not a major concern for you, could we merge the PR?

Reply:
Nope. Not a blocker for me.